src/endpoint/sts/ops/sts_post_assume_role.js

```diff
@@ -49,11 +52,52 @@ async function assume_role(req) {
 }
 
 // create and return the signed token
+/**
+ * @param {Object} auth_options
+ * @param {Number} expiry in seconds
+ * @returns {String}
+*/
 function generate_session_token(auth_options, expiry) {
     dbg.log1('sts_post_assume_role.make_session_token: ', auth_options, expiry);
     return jwt_utils.make_auth_token(auth_options, { expiresIn: expiry });
 }
 
+// TODO: Generalize and move to a utils file in the future
+/**
+ * @param {String|undefined} duration_input
+ * @returns {Number} duration in milliseconds
+ */
+function _parse_sts_duration(duration_input) {
+    if (duration_input === undefined) {
+        return config.STS_DEFAULT_SESSION_TOKEN_EXPIRY_MS;
+    }
+
+    const duration_sec = Number(duration_input);
+    if (!Number.isInteger(duration_sec)) {
+        throw new StsError(StsError.InvalidParameterValue);
+    }
+    if (duration_sec < config.STS_MIN_DURATION_SECONDS) {
+        throw new StsError(_sts_duration_validation_error(duration_input, 'greater', config.STS_MIN_DURATION_SECONDS));
+    }
+    if (duration_sec > config.STS_MAX_DURATION_SECONDS) {
+        throw new StsError(_sts_duration_validation_error(duration_input, 'less', config.STS_MAX_DURATION_SECONDS));
+    }
+
+    return duration_sec;
+}
+
+function _sts_duration_validation_error(duration_input, constraint, constraint_value) {
+    return {
+        ...StsError.ValidationError,
+        message: `Value ${duration_input} for durationSeconds failed to satisfy constraint: Member must have value ${constraint} than or equal to ${constraint_value}`,
+    };
+}
+
+function _calculate_expiration_time(seconds_until_expiration) {
+    const duration_ms = seconds_until_expiration * 1000;
+    return Math.ceil((Date.now() + duration_ms) / 1000);
```

Comment:
I don't understand why you divide by 1000 here? We are supposed to return a time from this function, so shouldn't it be in ms from epoc? it goes back to the previous comment - I think all times should be in ms to prevent confution, and only for specific functions that check for seconds we should convert - which in our case would be for generate_session_token(), and naturally also when parsing the duration input which is in seconds, but then I would expect to convert it to ms for working with it further.